lib/library.rb

```diff
@@ -0,0 +1,48 @@
+class Library
+  attr_reader :name, :books, :authors, :checked_out_books
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []    
+    @checked_out_books = []
+    @checkout_log = []
+  end
+
+  def add_author(author)
+    #an add_book method is probably more proper as a library is unlikely to add all books of an author at once.
```

Comment:
Also, what happens when the author writes a new book after they're added to the library? We'd need some way to account for that. If we act under the assumption that a library will hold every book from an author, then we can just calculate those books based off of the @authors array.